homeassistant/components/broadlink/__init__.py

```diff
@@ -35,7 +39,16 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         data.heartbeat = BroadlinkHeartbeat(hass)
         hass.async_create_task(data.heartbeat.async_setup())
 
-    device = BroadlinkDevice(hass, entry)
+    code_storage = Store(
+        hass, CODE_STORAGE_VERSION, f"broadlink_remote_{entry.unique_id}_codes"
+    )
+    flag_storage = Store(
+        hass, FLAG_STORAGE_VERSION, f"broadlink_remote_{entry.unique_id}_flags"
+    )
+    store = BroadlinkStores(code_storage, flag_storage)
+    await store.async_setup()
+
+    device = BroadlinkDevice(hass, entry, store)
```

Comment:
We need to check if the device is an universal remote before creating the code storage. This logic fits better in device.async_setup().

Reply:
Thinking better, is not a bad idea to have this here, decoupled, so we can optimize async_reload_entry() to prevent reloading the storage in the future. Just make sure to check if it is a remote.